### PR TITLE
[PLAT-4903] Update Maze-runner to V2

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

Updates Maze-runner to the latest version, with appropriate changes to the feature and fixture files.  This will allow us to bring proxy changes and other newer features from Maze-runner directly into this repo's tests. 

## Design

Bulk update of the fixtures, steps and scenarios to use Maze V2 steps and methods.

I'll outline a couple of places where changes have been made I'm not especially happy with, and will look into fixing either as a part of this PR or in a separate ticket.

### CODE REVIEW

File: Gemfile
Comment (Cawllec) [inline Gemfile:3]: This needs to be changed back once https://github.com/bugsnag/maze-runner/pull/125 has been merged/released.  I'm going to address a couple of rubocop things there first.

File: features/abi_apk_splits.feature
Comment (Cawllec) [inline features/abi_apk_splits.feature:6]: Several features like this are heavily dependent on payload ordering.  I intend to address this in a separate PR targeting this branch before this branch is merged.

File: features/abi_apk_splits.feature
Comment (Cawllec) [inline features/abi_apk_splits.feature:83]: Several scenarios are being skipped currently due to the payload ordering issue.  That will be addressed in the PR mentioned above.

File: features/fixtures/app/gradlew
Comment (Cawllec) [inline features/fixtures/app/gradlew:3]: @fractalwrench I'm assuming this shouldn't be checked in?

File: features/fixtures/config/bugsnag/all_disabled.gradle
Comment (Cawllec) [inline features/fixtures/config/bugsnag/all_disabled.gradle:4]: I've hard coded the mock server port, as while we can technically change it there is no practical way or reason to.

File: features/steps/gradle_plugin_steps.rb
Comment (Cawllec) [inline features/steps/gradle_plugin_steps.rb:59]: I'm not overly fond of this re-write, but can't think of a better way of getting the exit_code.  It's using the tools we've previously written, but just feels off.

File: features/fixtures/app/module/config/bugsnag/all_disabled.gradle
Comment (Cawllec) [inline features/fixtures/app/module/config/bugsnag/all_disabled.gradle:2]: @fractalwrench Several of these configs seem to be additions rather than updates.  I'm not entirely sure where they've come from, but if you can confirm they aren't needed I'll remove them.

File: features/app_bundle.feature
Comment (fractalwrench) [inline features/app_bundle.feature:3]: These are flaky tests and will need their request order sorting (just flagging this where I see it as I know it'll be addressed in a separate PR)

File: features/app_bundle.feature
Comment (fractalwrench) [inline features/app_bundle.feature:78]: Is there any way that we can avoid this wait?

File: features/disabled_bugsnag.feature
Comment (fractalwrench) [inline features/disabled_bugsnag.feature:5]: There are a couple of places which now have waits - why are they necessary?

File: features/fixtures/app/gradlew
Comment (fractalwrench) [reply features/fixtures/app/gradlew:3]: It's fine to check this in generally in JVM projects. The mazerunner tests update this file if the value of `GRADLE_WRAPPER_VERSION` is set however, which can mean files in the working tree change when the tests run.

File: features/fixtures/app/module/config/bugsnag/all_disabled.gradle
Comment (fractalwrench) [reply features/fixtures/app/module/config/bugsnag/all_disabled.gradle:2]: As part of v5 I believe the directory containing these snippets was moved to `features/fixtures/app/config` - so these should be safe to remove

**Review by fractalwrench — COMMENTED** [review]
Also noticed that a few scenarios are failing locally for me:

>cucumber features/app_bundle.feature:34 # Scenario: Bundling multiple flavors automatically
cucumber features/fail_on_upload_error.feature:3 # Scenario: Upload successfully with API key, mapping file, and correct endpoint
cucumber features/fail_on_upload_error.feature:8 # Scenario: No uploads or build failures when obfuscation is disabled
cucumber features/flavors.feature:3 # Scenario: Flavors automatic upload on build
cucumber features/ndk_app_agp400.feature:4 # Scenario: NDK apps send requests
cucumber features/ndk_app_agp400.feature:38 # Scenario: Custom projectRoot is added to payload
cucumber features/ndk_app_agp400.feature:66 # Scenario: Custom objdump location

File: features/app_bundle.feature
Comment (Cawllec) [reply features/app_bundle.feature:78]: Not really, we could shorten it to 1-3 seconds, but we need to allow a bit of a wait

File: features/disabled_bugsnag.feature
Comment (Cawllec) [reply features/disabled_bugsnag.feature:5]: They're needed to make sure we don't have a "late" payload coming through.  If we don't expect a payload, without a wait the likelihood of false positive passes increases greatly.

File: Gemfile
Comment (twometresteve) [reply Gemfile:3]: Any reason not to merge and release #125 now?

File: features/disabled_bugsnag.feature
Comment (twometresteve) [reply features/disabled_bugsnag.feature:5]: I think we tend to think about this more when expecting 0 requests, but not 1 or more (where the same issue of receiving too many requests could happen).  I guess we just consider 0 to be a special case more worthy of waiting for?  We do already have a ticket to address tests with no positive outcome, perhaps by capturing logs instead, so we can probably tolerate this for now.  I did think that we could put some functionality into Maze Runner itself to wait in just the 0 request case, but that's probably compounding the issue.

Comment (fractalwrench) [reply features/disabled_bugsnag.feature:5]: fair enough - I'm happy with adding a wait if it's necessary for the test and there are plans to rewrite tests which expect 0 requests anyway

File: Gemfile
Comment (twometresteve) [reply Gemfile:3]: But also, how do we want to track Maze Runner from BAGP when it's not using the Docker image?  Just pinning it to a release (tag) and not worrying about trying to track the latest at this stage?

File: features/steps/gradle_plugin_steps.rb
Comment (twometresteve) [reply features/steps/gradle_plugin_steps.rb:59]: Return the exit code from the method that invoked it seems perfectly reasonable to me.

**Review by twometresteve — COMMENTED** [review]
I've given this a static review having seen @fractalwrench 's comments, confident in this eagle eye.  I'm happy to merge this once he approves.

File: Gemfile
Comment (Cawllec) [reply Gemfile:3]: Pretty much.  We should pin it to a version or branch tag and only update when necessary.

**Review by fractalwrench — APPROVED** [review]

**Comment by fractalwrench** [comment]
This will require #299 before merging but LGTM otherwise.

**Pull request merged**
